include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1032,52 @@ namespace GridTools
   ( const MeshType                                                                    &mesh,
     const std::function<bool (const typename MeshType::active_cell_iterator &)> &predicate );
 
+  /**
+   * Compute a collection of bounding boxes so that all cells for which the given predicate is true, are
+   * completely enclosed in at least one of the bounding boxes.
+   *
+   * For each cell at a given refinement level containing active cells for which predicate is true,
+   * the function creates a bounding box of its children for which predicate is true.
+   *
+   * This results in a cover of these cells; the parameters @p allow_merge and @p max_boxes are used to
+   * reduce the number of cells at a computational cost and covering a bigger n-dimensional volume.
+   *
+   * The parameters to control the algorithm are:
+   * - @p predicate : the property of the cells to enclose e.g. IteratorFilters::LocallyOwnedCell .
+   *  The predicate is tested only on active cells.
+   * - @p refinement_level : it defines the level at which the initial bounding box are created. The refinement
+   *  should be set to a coarse refinement level. If @p refinement_level is higher than the number of levels of the
+   *  triangulation an exception is thrown.
+   * - @p allow_merge : This flag allows for box merging and, by default, is false. The algorithm has a cost of
+   *  O(N^2) where N is the number of the bounding boxes created from the refinement level; for this reason, if
+   *  the flag is set to true, make sure to choose wisely a coarse enough @p refinement_level.
```

Comment:
I don't understand why a coarser level gives fewer boxes. Maybe I misunderstand exactly what is going on.

Reply:
I'm sorry, I believe ffrom your comments I think you slightly misunderstood a part of the algorithm.
The idea is to set a refinement level: this is chosen. Now, for simplicity assume there are no active cells at refinement level or coarser levels. Then there shall be a maximum number of  bounding boxes which is given by the number of cells at this level.
What happen is: call C the a cell at this level. If it has some children which have the property set to true, then you create a box that:
- it's the smallest box inside C and containing all the active cells inside C for which the property is true
- it might possibly contain some more area, unless the geometry is regular

Now you shall have a set of these bounding boxes. You can see that if, instead of C I choose its parent, I shall obtain less Bounding Boxes and less accuracy (without thinking about the merging part).

Reply:
Okay, I think I finally got it.